test/e2e/jobs/jobs.test.ts

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2025 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { GenericContainer, StartedTestContainer, Wait } from "testcontainers";
+// import { LogWaitStrategy } from "testcontainers/build/wait-strategies/log-wait-strategy";
+import { CommunicationProtocolEnum, DaprClient, DaprServer } from "../../../src";
+// import { AbstractWaitStrategy } from "testcontainers/build/wait-strategies/wait-strategy";
+
+jest.setTimeout(10000);
+
+describe("Jobs End to End", () => {
+
+    let daprScheduler: StartedTestContainer | null = null;
+    let daprd: StartedTestContainer | null = null;
+    let server: DaprServer | null = null;
+    let client: DaprClient | null = null;
+
+    beforeAll(async () => {
+
+        daprScheduler = await new GenericContainer("ghcr.io/dapr/dapr")
+          .withExposedPorts(
```

Comment:
I'm looking at this to see how a real DaprContainer Testcontainer would be used.

Note: fixed host port bindings are not recommended

https://node.testcontainers.org/features/containers/#exposing-container-ports

Consider using a network alias

https://node.testcontainers.org/features/networking/#network-aliases

Reply:
What specifically needs to communicate with what?

Checkout the use of socat:

https://node.testcontainers.org/features/containers/#socatcontainer-as-a-tcp-proxy

Test:

https://github.com/testcontainers/testcontainers-node/blob/main/packages/testcontainers/src/socat/socat-container.test.ts